string color

---

int32 x_dev
int32 y_dev
